ci/pkg/ci/ci.go

```diff
@@ -0,0 +1,327 @@
+		logrus.Warn("Detected possible token in Git Origin, stripping git origin.")
+		originSplit := strings.Split(origin, "@")
+		// Take the substring after the last @ to avoid any tokens in an HTTPS URL
+		origin = originSplit[len(originSplit)-1]
+
+	}
+
+	url := fmt.Sprintf("%s/v0/organizations/%s/ci/scan-results", configurationObject.Options.Hostname, configurationObject.Options.Organization)
+	req, err := http.NewRequest("POST", url, &b)
+	if err != nil {
+		logrus.Warn("Unable to create Request")
+		return err
+	}
+
+	req.Header.Set("Content-Type", w.FormDataContentType())
+	req.Header.Set("X-Commit-Hash", currentHash)
+	req.Header.Set("X-Branch-Name", branchName)
+	req.Header.Set("X-Master-Hash", masterHash)
+	req.Header.Set("X-Repository-Name", origin)
+	req.Header.Set("Authorization", "Bearer "+token)
+
+	for _, report := range reports {
+		req.Header.Set("X-Fairwinds-Report-Version-"+report.Report, report.Version)
+	}
+
+	client := &http.Client{
+		CheckRedirect: func(req *http.Request, via []*http.Request) error {
+			return http.ErrUseLastResponse
+		},
+	}
+	resp, err := client.Do(req)
+	if err != nil {
+		logrus.Warn("Unable to Post results to Insights")
+		return err
+	}
+	if resp.StatusCode != http.StatusOK {
+		return fmt.Errorf("Invalid status code: %d", resp.StatusCode)
+	}
+
+	defer resp.Body.Close()
+	body, err := ioutil.ReadAll(resp.Body)
+	if err != nil {
+		logrus.Warn("Unable to read results")
+		return err
+	}
+	var results ScanResults
+	err = json.Unmarshal(body, &results)
+	if err != nil {
+		return err
+	}
+	for idx, actionItem := range results.ActionItems {
+		for _, resource := range resources {
+			if resource.Kind == actionItem.ResourceKind && resource.Name == actionItem.ResourceName {
+				results.ActionItems[idx].Notes = fmt.Sprintf("Resource was found in file: %s with comment of %s", resource.Filename, resource.FileComment)
+				break
+			}
+		}
+	}
+	if configurationObject.Options.JUnitOutput != "" {
+		cases := make([]formatter.JUnitTestCase, 0)
+
+		for _, actionItem := range results.ActionItems {
+			cases = append(cases, formatter.JUnitTestCase{
+				Name: actionItem.ResourceName + ": " + actionItem.Title,
+				Failure: &formatter.JUnitFailure{
+					Message:  actionItem.Remediation,
+					Contents: fmt.Sprintf("File: %s\nDescription: %s", actionItem.Notes, actionItem.Description),
+				},
+			})
+		}
+
+		testSuites := formatter.JUnitTestSuites{
+			Suites: []formatter.JUnitTestSuite{
+				{
+					Tests:     len(results.ActionItems),
+					TestCases: cases,
+				},
+			},
+		}
+
+		xmlBytes, err := xml.MarshalIndent(testSuites, "", "\t")
+		if err != nil {
+			return err
+		}
+		xmlBytes = append([]byte(xml.Header), xmlBytes...)
+		err = ioutil.WriteFile(configurationObject.Options.JUnitOutput, xmlBytes, 0644)
+		if err != nil {
+			return err
+		}
+	}
+	logrus.Infof("Score of %f with a baseline of %f", results.Score, results.BaselineScore)
```

Comment:
I'd suggest breaking out the results handling (and maybe the junit output) from the HTTP logic